src/cache/mod.rs

```diff
@@ -171,9 +172,7 @@ mod tests {
 
         let result = app_ctx
             .cache
-            .get_or_insert(get_key, &app_ctx, |_ctx: &AppContext| {
-                Ok("loco-cache-value".to_string())
-            })
+            .get_or_insert(get_key, async { Ok("loco-cache-value".to_string()) })
```

Comment:
can you try a test where the async block captures something from app_ctx? like try a real DB call for example

Reply:
Added endpoint usage example + adding tests